components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2178,25 +2171,27 @@ private UsersGetResponse getMultiAttributeFilteredUsers(Node node, Map<String, B
             paginatedUserResult = getMultiAttributeFilteredUsersWithMaxLimit(node, offset, sortBy, sortOrder, domainName, limit,
                     true);
         } else {
-            paginatedUserResult = getMultiAttributeFilteredUsersWithMaxLimit(node, offset, sortBy, sortOrder, domainName, maxLimit,
-                    false);
+            int maxLimit = getMaxLimit(domainName);
+            paginatedUserResult = getMultiAttributeFilteredUsersWithMaxLimit(node, offset, sortBy, sortOrder,
+                    domainName, maxLimit, false);
         }
         filteredUserDetails = getFilteredUserDetails(new LinkedHashSet<>(paginatedUserResult.getFilteredUsers()),
                 requiredAttributes);
 
+        int maxLimitForTotalResults = getMaxLimitForTotalResults(domainName);
```

Comment:
Use this value of total result calculation logic.
The previous variable is used for the user retrieval process as per line 2174 considering the max user list length of the given user store.